### PR TITLE
Switch from poetry to uv

### LINKED ISSUE #232: Switch from poetry to uv
https://github.com/astral-sh/uv

There's a start at it in  #226

### PR DESCRIPTION
**Pull request by pantew869**
Fixes #232

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Doing a test build now. Would also be nice to switch pyenv to uv.

File: userspace/pyproject.toml [hunk 1]
```diff
@@ -1,104 +1,113 @@
...
+    "pytest-subtests",
+    "pytest-timeout",
+    "pytest-xdist",
+    "python-dateutil",
+    "pyyaml",
+    "pyzmq",
+    "pyzstd",
+    "requests",
+    "scons",
+    "sconscontrib",
```
Comment (adeebshihadeh) [inline userspace/pyproject.toml:61]: make sure to use two spaces

File: userspace/pyproject.toml [hunk 2]
```diff
@@ -1,104 +1,113 @@
...
-pytest-xdist = "*"
-pytest-timeout = "*"
-pytest-random-order = "*"
-pytest-mock = "*"
-pytest-repeat = "*"
-pytest-randomly = "*"
-pytest-forked = "*"
+[project.optional-dependencies]
+dev = [
+    "ipython",
```
Comment (adeebshihadeh) [inline userspace/pyproject.toml:82]: same here

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -36,11 +36,3 @@ repos:
         - --select=F,E112,E113,E304,E502,E701,E702,E703,E71,E72,E731,W191,W6
         - --statistics
         - -j4
--   repo: https://github.com/python-poetry/poetry
```
Comment (adeebshihadeh) [inline .pre-commit-config.yaml:39]: is there a uv equivalent?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 3 above)
Comment (pantew869) [reply .pre-commit-config.yaml:39]: Afaik no

**Pull request merged**